server/index.js

```diff
@@ -276,6 +276,53 @@ app.get("/api/all-todos", async (req, res) => {
   }
 });
 
+/**
+ * REST API #8 for changing todo status
+ * @description
+ * - Extract todo id and todo status from request
+ * - Updates todo status in database
+ * - Gets the updated todo information
+ * - If success,
+ *  - Sends information of the modified todo
+ * - If fails,
+ *  - Sends 400 status
+ * @param {object} req HTTP request object
+ * @param {object} res HTTP response object
+ */
+app.post("/api/check-todo", async (req, res) => {
+  try {
+    const id = req.body.id;
+    const isDone = req.body.isDone;
+    const creatorId = await getUserIdFromRequest(req);
+
+    if (!creatorId) {
+      throw new Error();
+    }
```

Comment:
This will allow any logged in user to change any logged in user's todo.

An user should be able to change the status of a todo **ONLY IF** he is the creator of that todo.